pypeit/spectrographs/keck_nirspec.py

```diff
@@ -45,44 +45,113 @@ def __init__(self):
-        par['calibrations']['wavelengths']['sigdetect'] = 5.      # Good for NIRSPEC-1
+        par['calibrations']['wavelengths']['rms_threshold'] = 0.20 #0.20  # Might be grating dependent..
+        par['calibrations']['wavelengths']['sigdetect']=5.0
+        par['calibrations']['wavelengths']['fwhm']= 5.0
+        par['calibrations']['wavelengths']['n_final']= 4
+        par['calibrations']['wavelengths']['lamps'] = ['OH_NIRES']
+        par['calibrations']['wavelengths']['nonlinear_counts'] = self.detector[0]['nonlinear'] * self.detector[0]['saturation']
+        par['calibrations']['wavelengths']['method'] = 'holy-grail'
+        # Reidentification parameters
+        #par['calibrations']['wavelengths']['reid_arxiv'] = 'keck_nires.fits'
+        par['calibrations']['slitedges']['edge_thresh'] = 200.
+        par['calibrations']['slitedges']['sync_predict'] = 'nearest'
+
+        # Flats
+        par['calibrations']['flatfield']['tweak_slits_thresh'] = 0.80
+        par['calibrations']['flatfield']['illumflatten'] = True
+
+        # Extraction
+        par['scienceimage']['bspline_spacing'] = 0.8
+        par['scienceimage']['sn_gauss'] = 4.0
+
+        # Flexure
+        par['flexure']['method'] = 'skip'
 
+        par['scienceframe']['process']['sigclip'] = 20.0
+        par['scienceframe']['process']['satpix'] ='nothing'
+
+
+        # Overscan but not bias
+        #  This seems like a kludge of sorts
+        par['calibrations']['biasframe']['useframe'] = 'none'
+        # No overscan
+        par['scienceframe']['process']['overscan'] ='none'
+        for key in par['calibrations'].keys():
+            if 'frame' in key:
+                par['calibrations'][key]['process']['overscan'] = 'none'
+
+
+        # The settings below enable NIRSPEC dark subtraction from the traceframe and pixelflatframe, but enforce
+        # that this bias won't be subtracted from other images. It is a hack for now, because eventually we want to
+        # perform this operation with the dark frame class, and we want to attach individual sets of darks to specific
+        # images.
+        par['calibrations']['biasframe']['useframe'] = 'bias'
+        par['calibrations']['traceframe']['process']['bias'] = 'force'
+        par['calibrations']['pixelflatframe']['process']['bias'] = 'force'
+        par['calibrations']['arcframe']['process']['bias'] = 'skip'
+        par['calibrations']['tiltframe']['process']['bias'] = 'skip'
+        par['calibrations']['standardframe']['process']['bias'] = 'skip'
+        par['scienceframe']['process']['bias'] = 'skip'
+
+
+        # Set the default exposure time ranges for the frame typing
+        par['calibrations']['standardframe']['exprng'] = [None, 20]
+        par['calibrations']['arcframe']['exprng'] = [20, None]
+        par['calibrations']['darkframe']['exprng'] = [20, None]
+        par['scienceframe']['exprng'] = [20, None]
         return par
 
+    # JFH Replaced with values from MOSFIRE above.
```

Comment:
Not sure what happened here, but can you check that these changes were intentional?

Reply:
Yeah it was, but I'm keeping the old setting sin there for now.